app/src/main/java/com/ikarabulut/ComeAliveWithFour.java

```diff
@@ -1,17 +1,10 @@
 package com.ikarabulut;
 
-public class ComeAliveWithFour implements RuleSet {
+public class ComeAliveWithFour extends RuleSet {
 
-    public boolean willBeAliveNextGeneration(int aliveNeighborsCount, boolean isAlive) {
-        if (aliveNeighborsCount < 2) {
-            return isUnderpopulated();
-        } else if (isAlive && aliveNeighborsCount == 2
-                || isAlive && aliveNeighborsCount == 3
-                || !isAlive && aliveNeighborsCount == 4) {
-            return isFertileLand();
-        } else {
-            return isOverpopulated();
-        }
+    protected boolean isFertileLand(int aliveNeighborsCount, boolean isAlive) {
```

Comment:
This goes for all your rules classes: I think `isFertileLand` is doing too much. If I'm reading it correctly, it's not just dictating if a dead cell has enough live neighbors to come alive. It's also checking if an alive cell should stay alive. I think that should be in its own block so the logic is kept separate and it'd make it a lot easier to read.

Reply:
Oh interesting? Like a `shouldStayAlive()` and `isFertileLand()`?  If so, 👍

Reply:
Yup! I talked with @ikarabulut about it this morning but forgot to update here @jramnani . I think I would prefer the logic to be separated the same way it's separated when we talk about the rules of GoL (like in the README), My real issue here is that the terseness of the code makes it difficult to read, so even if we spread it out to a few if/else lines, that'd make it easier to read as another developer (or even Ish 2 weeks from now 🤣 )